File: src/modules/users/presenter/dtos/create_user_dto.py
from pydantic import BaseModel
from src.modules.shared.domain.entities import UserEntity


class CreateUserRequestDto(BaseModel):
    username: str
    email: str
    is_active: bool = True

    def to_entity(self) -> UserEntity:
        return UserEntity(username=self.username, email=self.email, is_active=self.is_active)

class CreateUserResponseDto(BaseModel):
    id: str
  

File: src/modules/users/presenter/dtos/create-user.dto.ts
import { IsBoolean, IsEmail, IsOptional, IsString } from 'class-validator';
import { UserEntity } from '../../../shared/domain/entities/user.entity';

export class CreateUserRequestDto {
  @IsString()
  username: string;

  @IsEmail()
  email: string;

  @IsBoolean()
  @IsOptional()
  isActive?: boolean = true;

  toEntity(): UserEntity {
    return new UserEntity({
      username: this.username,
      email: this.email,
      isActive: this.isActive,
    });
  }
}

export class CreateUserResponseDto {
  id: string;
}
